src/main/webapp/components/api/model-controller.service.js

```diff
@@ -36,22 +37,27 @@ angular.module('managementConsole.api')
             };
 
             /**
-             * Logs into the management endpoint and determines the launch type
+             * Logs into the management endpoint and determines the launch type.
+             * If username and/or password are not given, previously stored username and/or password are used.
              * @param {string} username - the username to use when connecting to the management endpoint
              * @param {string} password - the password to use when connecting to the management endpoint
              */
             ModelControllerClient.prototype.login = function (username, password) {
-                this.credentials.username = username;
-                this.credentials.password = password;
+                if (username !== undefined) { this.credentials.username = username; }
+                if (password !== undefined) { this.credentials.password = password; }
                 return this.readResource([], false, true).then(function (response) {
+                // TODO: Handle situation where response is received but login failed - that happens!
```

Comment:
When login fails we get an exception @Martinsos What I discovered is that we have to bind this in the catch block in order to properly logout - otherwise we never clean credentials from local storage